File: wear/src/main/java/com/gilcu2/gridpageviewer2/MyGridAdapter.java
package com.gilcu2.gridpageviewer2;

/**
 * Created by gilcu2 on 2/27/17.
 */

import android.content.Context;
import android.graphics.Color;
import android.support.wearable.view.GridPagerAdapter;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;


public class MyGridAdapter extends GridPagerAdapter {
    Context mContext;
    int[] mImages;

    public MyGridAdapter(Context context, int[] mImages) {
        mContext = context;
        this.mImages = mImages;
    }

    @Override
    public int getRowCount() {
        return 2;
    }

    @Override
    public int getColumnCount(int i) {

        if(i==0) return 3; else return 2;
    }

    //---Go to current column when scrolling up or down (instead of default column 0)---
    @Override
    public int getCurrentColumnForRow(int row, int currentColumn) {
        return currentColumn;
    }

    //---Return our car image based on the provided row and column---
    @Override
    public Object instantiateItem(ViewGroup viewGroup, int row, int col) {
        // Trick to use single dimensional array as multidimensional array.
        int imageToDisplay = 0;
        switch (row) {
            case 0:
                imageToDisplay = col;     // results are 0, 1, 2
                break;
            case 1:
                imageToDisplay = 3+ col;     // results are 3, 4, 5
                break;
        }
        ImageView imageView;
        imageView = new ImageView(mContext);
        imageView.setImageResource(mImages[imageToDisplay]);
        imageView.setBackgroundColor(Color.parseColor("#26ce61"));
        viewGroup.addView(imageView);
        return imageView;
    }

    @Override
    public void destroyItem(ViewGroup viewGroup, int i, int i2, Object o) {
        viewGroup.removeView((View) o);
    }

    @Override
    public boolean isViewFromObject(View view, Object o) {
        return view.equals(o);
    }
}

File: wear/src/main/java/com/gilcu2/gridpageviewer2/MyGridAdapter.kt
package com.gilcu2.gridpageviewer2

/**
 * Created by gilcu2 on 2/27/17.
 */

import android.content.Context
import android.graphics.Color
import android.support.v4.content.ContextCompat
import android.support.wearable.view.GridPagerAdapter
import android.util.TypedValue
import android.view.Gravity
import android.view.TextureView
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.learnandroidwear.simplegridview.R


class MyGridAdapter(internal var mContext: Context, internal var mImages: IntArray) : GridPagerAdapter() {

    override fun getRowCount(): Int {
        return 1
    }

    override fun getColumnCount(i: Int): Int {
        return 2
    }

    //---Go to current column when scrolling up or down (instead of default column 0)---
    override fun getCurrentColumnForRow(row: Int, currentColumn: Int): Int {
        return currentColumn
    }

    //---Return our car image based on the provided row and column---
    override fun instantiateItem(viewGroup: ViewGroup, row: Int, col: Int): Any {
        if (col == 0)
            return createPage1(viewGroup)
        else
            return createPage2(viewGroup)
    }

    override fun destroyItem(viewGroup: ViewGroup, i: Int, i2: Int, o: Any) {
        viewGroup.removeView(o as View)
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return view == o
    }

    fun createPage1(viewGroup: ViewGroup): Any {

        val layout = LinearLayout(mContext)
        layout.setOrientation(LinearLayout.VERTICAL)

        val text = TextView(mContext)
        text.text = "1"
        text.gravity = Gravity.CENTER or Gravity.TOP
        text.setTextSize(TypedValue.COMPLEX_UNIT_SP, 80f)
        layout.addView(text)

        val layoutB = LinearLayout(mContext)

        val buttonL = ImageButton(mContext)
        buttonL.setImageDrawable(ContextCompat.getDrawable(mContext, R.drawable.ic_brightness_4_black_24dp))
        val size = (40 * mContext.getResources().getDisplayMetrics().density + 0.5f).toInt()
        val params = ViewGroup.LayoutParams(size, size)
        buttonL.layoutParams = params
//        button.foregroundGravity = Gravity.LEFT or Gravity.BOTTOM
        layoutB.addView(buttonL)

        val buttonR = ImageButton(mContext)
        buttonR.setImageDrawable(ContextCompat.getDrawable(mContext, R.drawable.ic_brightness_5_black_24dp))
        buttonR.layoutParams = params
//        button.foregroundGravity = Gravity.LEFT or Gravity.BOTTOM
        layoutB.addView(buttonR)

        layout.addView(layoutB)


        viewGroup.addView(layout)
        return layout
    }

    fun createPage2(viewGroup: ViewGroup): Any {
        val text = TextView(mContext)
        text.text = "2"
        viewGroup.addView(text)
        return text
    }
}
